internal/operator-controller/controllers/clusterextensionrevision_controller.go

```diff
@@ -605,3 +517,66 @@ var (
 		FieldB: ".status.replicas",
 	}
 )
+
+func setRetryingConditions(cer *ocv1.ClusterExtensionRevision, message string) {
+	markAsProgressing(cer, ocv1.ClusterExtensionRevisionReasonRetrying, message)
+	if meta.FindStatusCondition(cer.Status.Conditions, ocv1.ClusterExtensionRevisionTypeAvailable) != nil {
+		markAsAvailableUnknown(cer, ocv1.ClusterExtensionRevisionReasonReconciling, message)
+	}
+}
+
+func markAsProgressing(cer *ocv1.ClusterExtensionRevision, reason, message string) {
+	meta.SetStatusCondition(&cer.Status.Conditions, metav1.Condition{
+		Type:               ocv1.ClusterExtensionRevisionTypeProgressing,
+		Status:             metav1.ConditionTrue,
+		Reason:             reason,
+		Message:            message,
+		ObservedGeneration: cer.Generation,
+	})
+}
+
+func markAsNotProgressing(cer *ocv1.ClusterExtensionRevision, reason, message string) bool {
+	return meta.SetStatusCondition(&cer.Status.Conditions, metav1.Condition{
+		Type:               ocv1.ClusterExtensionRevisionTypeProgressing,
+		Status:             metav1.ConditionFalse,
+		Reason:             reason,
+		Message:            message,
+		ObservedGeneration: cer.Generation,
+	})
+}
+
+func markAsAvailable(cer *ocv1.ClusterExtensionRevision, reason, message string) bool {
+	return meta.SetStatusCondition(&cer.Status.Conditions, metav1.Condition{
+		Type:               ocv1.ClusterExtensionRevisionTypeAvailable,
+		Status:             metav1.ConditionTrue,
+		Reason:             reason,
+		Message:            message,
+		ObservedGeneration: cer.Generation,
+	})
+}
+
+func markAsUnavailable(cer *ocv1.ClusterExtensionRevision, reason, message string) {
+	meta.SetStatusCondition(&cer.Status.Conditions, metav1.Condition{
+		Type:               ocv1.ClusterExtensionRevisionTypeAvailable,
+		Status:             metav1.ConditionFalse,
+		Reason:             reason,
+		Message:            message,
+		ObservedGeneration: cer.Generation,
+	})
+}
+
+func markAsAvailableUnknown(cer *ocv1.ClusterExtensionRevision, reason, message string) bool {
+	return meta.SetStatusCondition(&cer.Status.Conditions, metav1.Condition{
+		Type:               ocv1.ClusterExtensionRevisionTypeAvailable,
+		Status:             metav1.ConditionUnknown,
+		Reason:             reason,
+		Message:            message,
+		ObservedGeneration: cer.Generation,
+	})
+}
+
+func setArchivedConditions(cer *ocv1.ClusterExtensionRevision) bool {
```

Comment:
[nit] perhaps to rename to `markAsArchived` to match the `markAs*` functions?

Reply:
Done